js/wordlist.js

```diff
@@ -12,11 +19,45 @@ let showZhuyin = (localStorage['zhuyin'] === 'yes');
 let entries;
 if (wordList) {
     entries = JSON.parse(wordList);
-    entries.forEach(e => { e.notes = (e.notes || '<i>Edit</i>');});
+    let missingZhuyin = false;
+    entries.forEach(e => { 
+        e.notes = (e.notes || '<i>Edit</i>');
+
+        if (!e.zhuyin) {
+            missingZhuyin = true;
+            let pinyin = e.pinyin.split(/[\s·]+/);
+            let syllables = [];
+            /* search for tones */
+            pinyin.forEach(py => {
+                py = py.toLowerCase();
+                let tone = getToneNumber(py);
+                let syllable = removeToneMarker(py, tone) + tone;
```

Comment:
I feel like these to calls could be combined in one function (`pinyinToSyllable` for example).
Right now it was a bit confusing to read.